src/main/features/claim/form/models/claimAmountBreakdown.ts

```diff
@@ -48,7 +43,17 @@ export default class ClaimAmountBreakdown implements Serializable<ClaimAmountBre
   }
 
   appendRow () {
-    this.rows.push(ClaimAmountRow.empty())
+    if (this.rows.length < MAX_NUMBER_OF_ROWS) {
+      this.rows.push(ClaimAmountRow.empty())
+    }
+  }
+
+  removeExcessRows () {
+    this.rows = this.rows.filter(item => !!item.amount && !!item.reason)
+
+    if (this.rows.length === 0) {
+      this.appendRow()
+    }
   }
 
   totalAmount () {
```

Comment:
the calculation below here should be done with a `reduce` function

Reply:
you love reduce, don't you ;)